spec/abstract/KHeap_A.thy

```diff
@@ -216,6 +226,15 @@ abbreviation
 
 section \<open>Scheduling Contexts\<close>
 
+definition
+  read_sched_context :: "obj_ref \<Rightarrow> (sched_context,'z::state_ext) r_monad"
+where
+  "read_sched_context ptr \<equiv> do {
+     kobj \<leftarrow> read_object ptr;
+     case kobj of SchedContext sc n \<Rightarrow> oreturn sc
```

Comment:
Like for tcbs, I'd factor out the `case`, which enables both projection and reader, i.e.

`sc_of obj = case obj of SchedContext sc n => Some sc | _ => None`

This one is a bit more non-standard, because we are discarding the `n`, but I think the version where we'd want to keep that would be `sc_n_of`, so that's fine.

With that you can have the same pattern as above:
```
abbreviation
scs_of s = kheap s |> sc_of

read_sched_context = swp scs_of
```